src/thistlethwaite.rs

```diff
@@ -19,105 +21,139 @@ pub fn solve(cube: &impl Cube) -> Option<Vec<Move>> {
+                Move::F(Double), Move::B(Double) 
+            ]
+        };
+
+        static ref PRUNING_TABLE: PruningTable = {
+            let pruning_depth = 5;
+            PruningTable::new(&[FaceletCube::new(3).mask(&*MASK)], pruning_depth, &*MOVES)
+        };
+
+        static ref SOLVER: Solver = {
+            Solver::new((*MOVES).clone(), (*PRUNING_TABLE).clone())
+        };
+    }
 
-    ida_star(&cube.mask(&mask), &solver, search_limit)
+    ida_star(&cube.mask(&*MASK), &*SOLVER, 10)
 }
 
 pub fn phase3(cube: &impl Cube) -> Option<Vec<Move>>  {
     use Face::*;
     use MoveVariant::*;
 
-    let cp_pieces = [U, D, F, B, L, R].iter()
-                                      .map(|f| [1, 3, 7, 9].map(|x| S(3, *f, x)))
-                                      .collect::<Vec<_>>()
-                                      .concat();
-
-    let ep_pieces = [F, B, L, R].iter()
-                                .map(|f| [2, 4, 6, 8].map(|x| S(3, *f, x)))
-                                .collect::<Vec<_>>()
-                                .concat();
-
-    let face = |f| if f == B { F }
-                   else if f == L { R }
-                   else { f };
-
-    let mask = |i: i32, _| if cp_pieces.contains(&i) { [U, R, F, D, L, B][(i / 9) as usize] }
-                           else if ep_pieces.contains(&i) { face([U, R, F, D, L, B][(i / 9) as usize]) }
-                           else { X };
-
-    let moves = vec![
-        Move::U(Standard), Move::U(Inverse), Move::U(Double),
-        Move::D(Standard), Move::D(Inverse), Move::D(Double),
-        Move::F(Double), Move::B(Double), Move::L(Double), Move::R(Double)
-    ];
-
-    let solved_states_viewed_in_g2 = PruningTable::new(
-        &[FaceletCube::new(3).mask(&mask)], 
-        10,
-        &vec![Move::U(Double), Move::D(Double), Move::F(Double), Move::B(Double), Move::L(Double), Move::R(Double)]
-    );
-
-    let search_limit = 13;
-    let pruning_depth = 5;
-    let pruning_table = PruningTable::from_existing_table(&solved_states_viewed_in_g2, pruning_depth, &moves);
-
-    let solver = Solver::new(moves, pruning_table);
+    lazy_static! {
+        static ref MASK: Box<dyn Fn(i32, Face) -> Face + Sync> = {
+            let cp_pieces = [U, D, F, B, L, R].iter()
+                .map(|f| [1, 3, 7, 9].map(|x| S(3, *f, x)))
+                .collect::<Vec<_>>()
+                .concat();
+
+            let ep_pieces = [F, B, L, R].iter()
+                .map(|f| [2, 4, 6, 8].map(|x| S(3, *f, x)))
+                .collect::<Vec<_>>()
+                .concat();
+
+            let face = |f| if f == B { F }
+                           else if f == L { R }
+                           else { f };
+
+            Box::new(move |i: i32, _| 
+                    if cp_pieces.contains(&i) { [U, R, F, D, L, B][(i / 9) as usize] }
+                    else if ep_pieces.contains(&i) { face([U, R, F, D, L, B][(i / 9) as usize]) }
+                    else { X }
```

Comment:
Indentation different to line 74. Line 74 is preferred?

Reply:
Fixed.